components/file_limit_sticky_banner/index.tsx

```diff
@@ -0,0 +1,185 @@
+        const snoozeCoolOffDaysMillis = snoozeCoolOffDays * 24 * 60 * 60 * 1000;
+        const timeDiff = Date.now() - snoozeInfo.LastSnoozeTimestamp;
+        shouldShowAgain = timeDiff >= snoozeCoolOffDaysMillis;
+    }
+
+    if (!show) {
+        return null;
+    }
+
+    if (!shouldShowAgain) {
+        return null;
+    }
+
+    if (!isCloud || !isStarter) {
+        return null;
+    }
+
+    let isAbovePlanFileStorageLimit = false;
+
+    if ((limitsLoaded || hasSomeLimits(cloudLimits))) {
+        const currentFileLimitGB = cloudLimits.files?.total_storage || fallbackStarterLimits.files.totalStorage;
+        if (currentFileLimitGB > maxStarterPlanFileStorageGB) {
+            isAbovePlanFileStorageLimit = true;
+        }
+    }
+
+    if (isAbovePlanFileStorageLimit) {
+        return null;
+    }
+
+    const snoozeBanner = () => {
+        const fileLimitBannerSnoozeInfo = {
+            LastSnoozeTimestamp: Date.now(),
+        };
+
+        dispatch(savePreferences(user.id, [
+            {
+                category: Preferences.CLOUD_USER_EPHEMERAL_INFO,
+                name: 'file_limit_banner_snooze',
+                user_id: user.id,
+                value: JSON.stringify(fileLimitBannerSnoozeInfo),
+            },
+        ]));
+
+        setShow(false);
+    };
+
+    const AdminMessageLink = (
+        <a
+            onClick={(e) => {
+                e.preventDefault();
+                openPricingModal();
+            }}
+        >
+            {formatMessage({id: 'create_post.file_limit_sticky_banner.admin_link_message', defaultMessage: 'upgrade to a paid plan.'})}
+        </a>
+    );
+
+    const AdminMessage = (
+        <span>
+            {formatMessage({
+                id: 'create_post.file_limit_sticky_banner.admin_message',
+                defaultMessage: 'Your free plan is limited to {storageGB}GB of files. New uploads will automatically archive older files. To view them again, you can delete older files or '},
+            {
+                storageGB: '10',
+            },
+            )}
+            {AdminMessageLink}
+        </span>
+    );
+
+    const NonAdminMessage = (
+        <span>
+            {formatMessage({
+                id: 'create_post.file_limit_sticky_banner.non_admin_message',
+                defaultMessage: 'Your free plan is limited to {storageGB}GB of files. New uploads will automatically archive older files. To view them again, '},
+            {
+                storageGB: '10',
+            },
+            )}
+            {<NotifyAdminCTA CtaText={formatMessage({id: 'create_post.file_limit_sticky_banner.non_admin_link_message', defaultMessage: 'notify your admin to upgrade to a paid plan.'})}/>}
+        </span>
+    );
+
+    const tooltip = (
+        <Tooltip id='file_limit_banner_snooze'>
+            {formatMessage({id: 'create_post.file_limit_sticky_banner.snooze_tooltip', defaultMessage: 'Snooze for {snoozeDays} days'}, {snoozeDays: '10'})}
```

Comment:
Consider using the const here (which if you do, then we may need two consts, one for days, one for the millis)